lib/kafka/connection_builder.rb

```diff
@@ -10,11 +9,14 @@ def initialize(client_id:, logger:, instrumenter:, connect_timeout:, socket_time
       @connect_timeout = connect_timeout
       @socket_timeout = socket_timeout
       @ssl_context = ssl_context
-      @sasl_gssapi_principal = sasl_gssapi_principal
-      @sasl_gssapi_keytab = sasl_gssapi_keytab
-      @sasl_plain_authzid = sasl_plain_authzid
-      @sasl_plain_username = sasl_plain_username
-      @sasl_plain_password = sasl_plain_password
+      @authenticator = Authenticator.new(
+        sasl_gssapi_principal: sasl_gssapi_principal,
+        sasl_gssapi_keytab: sasl_gssapi_keytab,
+        sasl_plain_authzid: sasl_plain_authzid,
+        sasl_plain_username: sasl_plain_username,
+        sasl_plain_password: sasl_plain_password,
+        logger: @logger
+      )
```

Comment:
Can you move this initialization of Authenticator to Client and pass in the object to ConnectionBuilder as a parameter? That way we don't need to ferry all these arguments around.